Index: beast-0.7.1/birnet/birnettests.h
===================================================================
--- beast-0.7.1.orig/birnet/birnettests.h	2006-11-03 15:17:31.000000000 +0100
+++ beast-0.7.1/birnet/birnettests.h	2008-01-27 22:40:44.000000000 +0100
@@ -27,6 +27,7 @@
 
 #include <glib.h>
 #include <string.h>
+#include <signal.h>	/* G_BREAKPOINT() */
 
 BIRNET_EXTERN_C_BEGIN();
 
Index: beast-0.7.1/tools/bseloopfuncs.c
===================================================================
--- beast-0.7.1.orig/tools/bseloopfuncs.c	2008-01-27 22:40:24.000000000 +0100
+++ beast-0.7.1/tools/bseloopfuncs.c	2008-01-27 22:40:36.000000000 +0100
@@ -18,7 +18,7 @@
  */
 #include "bseloopfuncs.h"
 #include <bse/gsldatacache.h>
-#include <string.h>
+#include <signal.h>	/* G_BREAKPOINT() */
 #include <stdio.h>
 #include <math.h>
 
