diff -urNad beast-0.7.1.orig/birnet/mksignals.sh beast-0.7.1/birnet/mksignals.sh
--- beast-0.7.1.orig/birnet/mksignals.sh	2008-05-04 00:18:11.000000000 +0100
+++ beast-0.7.1/birnet/mksignals.sh	2008-05-04 00:19:06.000000000 +0100
@@ -5,7 +5,7 @@
 MAXIMUM="$2"
 [ -z "$MAXIMUM" ] && MAXIMUM="999999"
 
-function lineecho() {
+lineecho() {
 	# comment out the following line to get real line number information
 	echo "# $1 \"$2\""
 	echo # eat line 0
